File: python/leetcode/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_sum = -math.inf
        current_sum = 0

        for num in nums:
            if current_sum < 0:
                current_sum = 0
            current_sum += num
            max_sum = max(current_sum, max_sum)

        return max_sum

File: c#/leetcode/53-maximum-subarray.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
        
        if (nums.Length == 1) {
            return nums[0];
        }
        int largestSum = nums[0];
        int currentSum = nums[0];
        
        for (int i=1; i < nums.Length; i++) {
            if (currentSum < 0) {
                currentSum = 0;
            }
            currentSum += nums[i];
            largestSum = Math.Max(currentSum, largestSum);
        }
    return largestSum;
    }
}